index.d.ts

```diff
@@ -0,0 +1,173 @@
+declare module "bw-chk" {
+    type NameOrUnknown<T> = T | "Unknown";
+    type PossiblyUndefined<T> = {
```

Comment:
nit: This is functionally equivalent to the built-in `Partial<T>` type

Reply:
The difference is that the key must be defined and assigned to `undefined` where as Partial the key may be missing completely. The former matches the implementation, however either works.

Reply:
Yeah, feels a bit too specific to how the code is currently, would prefer to use the built-in type